### DISCUSSION TITLE
Displaying instances of programmatic workflows in the Dashboard

### DISCUSSION THREAD

**Discussion by glorious-beard** [elsa-workflows/elsa-core#3911] [Q&A]
Hi,

Do the instances in the Elsa dashboard only display those created from workflows created in the designer or JSON files? I have a programmatic workflow that I can see the instance of when I query `IWorkflowInstanceStore`, but I never see instances of my programmatic workflows on the dashboard.

**Comment by sfmskywalker** [comment]
Yes, all workflow instances should appear in the list, although I’m not sure if it’s fully supported yet in Elsa 3 as of yet (if not, it will be).

**Reply by glorious-beard** [reply]
Thanks,

I'm using Elsa 2.11.1; does that make a difference?

**Reply by sfmskywalker** [reply]
You should see instances of programmatic workflows in the dashboard. Since you mentioned you don’t see them show up, I think that’s a bug.

**Comment by martikyan** [comment]
What are the steps for converting `WorkflowBase` to `WorkflowDefinition` so that I can store it in `IWorkflowDefinitionStore`? I am trying to add the programmatic workflow to the UI.

Thanks

**Reply by sfmskywalker** [reply]
Elsa already converts coded workflows to `WorkflowDefinition` instances and then stores them in `IWorkflowDefinitionStore`, so you don't have to do this yourself. But if you wanted to, take a look at [ClrWorkflowProvider](https://github.com/elsa-workflows/elsa-core/blob/v3/src/modules/Elsa.Workflows.Runtime/Providers/ClrWorkflowProvider.cs) and [DefaultWorkflowDefinitionStorePopulator](https://github.com/elsa-workflows/elsa-core/blob/v3/src/modules/Elsa.Workflows.Runtime/Services/DefaultWorkflowDefinitionStorePopulator.cs).

The `DefaultWorkflowDefinitionStorePopulator` calls upon all `IWorkflowProvider` implementations (including `ClrWorkflowProvider`) to list their workflows, after which it will turn them into `WorkflowDefinition`s and store them in the database.

**Reply by martikyan** [reply]
Thanks for the reply @sfmskywalker 

I noticed that when listing definitions through UI there's a filter applied to `MaterializerName` to be `Json` but programmatically added workflows have `CLR` instead. Is it why they are not visible in the UI? Is that expected?

**Reply by sfmskywalker** [reply]
Indeed - the (Stencil) designer is only capable of displaying Flowchart activities for the moment, with Sequence support coming later.

That being said, we still want to be able to look at workflow instances and their journal from the dashboard.
For this reason, the Blazor implementation of the dashboard now _does_ support listing CLR workflows. This change was committed very recently, so you'll want to make sure to use the latest Elsa.Studio packages. Since the Stencil implementation is deprecated, it will not be updated with this support.
